services/messageService.js

```diff
@@ -0,0 +1,51 @@
+import { AsyncStorage } from 'react-native';
+
+class MessageService {
+  constructor(newMessagesCallback) {
+    this.messages = {};
+    this.newMessagesCallback = newMessagesCallback;
+  }
+
+  static async cacheMessages(contact, messages) {
+    try {
+      const localMessages = await AsyncStorage.getItem('messages');
+
+      let localMessageDict = {};
+      if (localMessages) {
+        localMessageDict = JSON.parse(localMessages);
+      }
+
+      localMessageDict[contact] = messages;
+      await AsyncStorage.setItem('messages', JSON.stringify(localMessageDict));
+    } catch (e) {
+      // TODO: Add sentry logging
+    }
+  }
+
+  async updateLocalMessageStateSingle(contact, message) {
+    const messages = {
+      items: [message]
+    };
+
+    this.updateLocalMessageState(contact, messages);
```

Comment:
this.updateLocalMessageState(contact, { items: [message] } );

Reply:
Thanks, fixed in next push.